modules/fedimint-ln/src/db.rs

```diff
@@ -121,140 +121,221 @@ impl_db_lookup!(
 
 #[cfg(test)]
 mod fedimint_migration_tests {
-    use std::collections::BTreeMap;
     use std::env;
     use std::path::Path;
+    use std::str::FromStr;
+    use std::time::SystemTime;
 
-    use fedimint_core::core::LEGACY_HARDCODED_INSTANCE_ID_LN;
-    use fedimint_core::db::apply_migrations;
+    use bitcoin_hashes::Hash;
+    use fedimint_core::db::{apply_migrations, Database};
     use fedimint_core::module::DynModuleGen;
-    use fedimint_testing::apply_to_databases;
+    use fedimint_core::{OutPoint, TransactionId};
+    use fedimint_testing::{open_temp_db, validate_migrations};
     use futures::StreamExt;
+    use rand::distributions::Standard;
+    use rand::prelude::Distribution;
+    use rand::rngs::OsRng;
+    use rand::RngCore;
     use strum::IntoEnumIterator;
+    use threshold_crypto::{DecryptionShare, G1Projective};
+    use url::Url;
 
+    use super::{
+        AgreedDecryptionShareKey, ContractKey, ContractUpdateKey, LightningGatewayKey, OfferKey,
+        ProposeDecryptionShareKey,
+    };
+    use crate::contracts::account::AccountContract;
+    use crate::contracts::incoming::{IncomingContractOffer, OfferId};
+    use crate::contracts::{
+        self, ContractId, EncryptedPreimage, Preimage, PreimageDecryptionShare,
+    };
     use crate::db::{
         AgreedDecryptionShareKeyPrefix, ContractKeyPrefix, ContractUpdateKeyPrefix, DbKeyPrefix,
-        LightningGatewayKeyPrefix, ProposeDecryptionShareKeyPrefix,
+        LightningGatewayKeyPrefix, OfferKeyPrefix, ProposeDecryptionShareKeyPrefix,
     };
-    use crate::LightningGen;
+    use crate::{ContractAccount, LightningGateway, LightningGen, LightningOutputOutcome};
+
+    async fn create_db_with_v0_data(db: Database) -> anyhow::Result<()> {
```

Comment:
Function populating the empty db with data should probably take a `DatabaseTransaction` just to cut on the common code.

Reply:
Done